core/src/main/resources/hudson/model/Job/configure.jelly

```diff
@@ -46,7 +46,7 @@ THE SOFTWARE.
 
             <j:if test="${it.isNameEditable()}">
               <f:entry title="${%name(it.pronoun)}">
-                <f:textbox name="name" value="${it.name}" />
+                <f:textbox name="name" value="${it.name}" autocomplete="on"/>
```

Comment:
Job configurations are definetely not safe, but the `name` field should be since it's not dynamically loaded in the same way as for example the snippet in a build step or JobProperty.